source/fundamentals/crud/write-operations/upsert.txt

```diff
@@ -4,8 +4,102 @@ Insert or Update in a Single Operation
 
 .. default-domain:: mongodb
 
-If your application stores and modifies data in MongoDB, you probably use
-insert and update operations. In certain workflows, you may need to choose
-between an insert and update depending on whether the document exists.
-In these cases, you can streamline your application logic by using the
-``upsert`` option available in the following methods:
+Overview
+--------
+
+You can insert or update documents by passing a query filter to the 
+:java-docs:`updateOne() </apidocs/mongodb-driver-sync/com/mongodb/client/MongoCollection.html#updateOne(org.bson.conversions.Bson,org.bson.conversions.Bson,com.mongodb.client.model.UpdateOptions)>`, 
+:java-docs:`updateMany() </apidocs/mongodb-driver-sync/com/mongodb/client/MongoCollection.html#updateMany(org.bson.conversions.Bson,org.bson.conversions.Bson,com.mongodb.client.model.UpdateOptions)>`
+or :java-docs:`replaceOne() </apidocs/mongodb-driver-sync/com/mongodb/client/MongoCollection.html#replaceOne(org.bson.conversions.Bson,TDocument,com.mongodb.client.model.ReplaceOptions)>` methods
+
+Each method returns an ``AcknowledgedUpdateResult`` that tells us:
+
+- The number of documents that matched the query filter
+- The number of documents that got modified 
+- The ``_id`` of the document that got upserted 
+
+These methods update documents that match your query filter, but
+inserts a document if the there are no matches when you set the upsert
+to true.  
```

Comment:
This still feels off. You state that documents can be inserrted with update or replace, but don't state how until after what the methods return. Additionally, is the information on what the methods return necessary here? Placing it so high implies it is important, but it doesn't seem that important.

Is this about what the methods return or about upsert behavior?